### ISSUE TITLE
Kaniko image for arm64 (aarch64) not pulling down anymore.

### ISSUE THREAD

**Issue by PREngineer** [GoogleContainerTools/kaniko#2678]
**Issue**
My apologies if this should not be asked here.  I don't know where to look for information.

I have a k3s cluster with Raspberry Pi 4s.  I have some Jenkins pipelines that were running fine until yesterday.  I noticed that all of them started failing with:

``
ERROR: Unable to pull Docker image "gcr.io/kaniko-project/executor:debug". Check if image tag name is spelled correctly.
``

Logged events say:

``
Failed to pull image "gcr.io/kaniko-project/executor:debug": rpc error: code = NotFound desc = failed to pull and unpack image "gcr.io/kaniko-project/executor:debug": no match for platform in manifest: not found
``

I'm wondering if this is an issue with Google Container Registry or if it could be my Pis?  Nothing in the pipeline code and Dockerfile has changed.

**Actual behavior**
Since yesterday, when building, it fails to pull the arm64/aarch64 version of the kaniko executor:debug.

**Expected behavior**
Kubernetes should be able to pull the image as before.

**To Reproduce**
1. Create Jenkins Pipeline, using Kaniko in Raspberry Pi4 (arm64/aarch64)
2. Run it

**Additional Information**
 - Kaniko Image: gcr.io/kaniko-project/executor:debug

**Comment by aaron-prindle** [comment]
Assuming the Raspberry Pi4 capable of using arm64 images, there is an arm64 image available at gcr.io/kaniko-project/executor:debug

command output showing the architectures available at `gcr.io/kaniko-project/executor:debug`
```
aprindle@aprindle-ssd ~/kaniko  [main]docker buildx imagetools inspect gcr.io/kaniko-project/executor:debug
Name:      gcr.io/kaniko-project/executor:debug
MediaType: application/vnd.oci.image.index.v1+json
Digest:    sha256:5e5214e7ec4a90569a2a660b1c7c3a9dafa8c22ede9a06851733cec5f42f4632
           
Manifests: 
  Name:      gcr.io/kaniko-project/executor:debug@sha256:a148912157a8b3b05f86070a6203912c91ccf067b1c61f112a6c6739da5e5f88
  MediaType: application/vnd.oci.image.manifest.v1+json
  Platform:  linux/amd64
             
  Name:      gcr.io/kaniko-project/executor:debug@sha256:0bb82563805003a78abd2bf700434005b9ef5d588ddfcd45054d0a32e4f5a13a
  MediaType: application/vnd.oci.image.manifest.v1+json
  Platform:  linux/arm64 # < -------------- arm64 image
             
  Name:      gcr.io/kaniko-project/executor:debug@sha256:d7e5ef274fb7658f95757193b89f76746a30854acf900dfc8ae7107348374708
  MediaType: application/vnd.oci.image.manifest.v1+json
  Platform:  linux/s390x
             
  Name:      gcr.io/kaniko-project/executor:debug@sha256:560959ff5854677baf1043ae4b210845e1a618cfdff7493abd42ed04d44a5139
  MediaType: application/vnd.oci.image.manifest.v1+json
  Platform:  unknown/unknown
    vnd.docker.reference.digest: sha256:a148912157a8b3b05f86070a6203912c91ccf067b1c61f112a6c6739da5e5f88
    vnd.docker.reference.type:   attestation-manifest
             
  Name:      gcr.io/kaniko-project/executor:debug@sha256:38e52f07e028a38b2df14318832440987ca8bddc6f99c972d6ce8219eab6ca52
  MediaType: application/vnd.oci.image.manifest.v1+json
  Platform:  unknown/unknown
    vnd.docker.reference.digest: sha256:0bb82563805003a78abd2bf700434005b9ef5d588ddfcd45054d0a32e4f5a13a
    vnd.docker.reference.type:   attestation-manifest
             
  Name:      gcr.io/kaniko-project/executor:debug@sha256:62149c06a76a2ec7569908229396e79257bcfa428410a697688bfb1e18254275
  MediaType: application/vnd.oci.image.manifest.v1+json
  Platform:  unknown/unknown
    vnd.docker.reference.digest: sha256:d7e5ef274fb7658f95757193b89f76746a30854acf900dfc8ae7107348374708
    vnd.docker.reference.type:   attestation-manifest
```

The code that uploads images to `gcr.io/kaniko-project/executor:debug` (showing the supported platforms) is here:
https://github.com/GoogleContainerTools/kaniko/blob/main/.github/workflows/images.yaml#L43C1-L43C1

and it shows the supported platforms for the `gcr.io/kaniko-project/executor:debug` image:
```
platforms: linux/amd64,linux/arm64,linux/s390x
```

**Comment by PREngineer** [comment]
Hi Aaron,

Thanks for replying.  My understanding is that it is using the linux/arm64 image tag.  The problem started yesterday, that's why I was wondering if it was a GCR issue.  When I visited gcr.io/kaniko-project in my browser, I saw a warning:

**Failed to load the list of repositories for the following hosts: us.gcr.io, asia.gcr.io, eu.gcr.io**

Problems started for me in Aug/10/2020 9:44:06 PM.  Last successful run was Aug/9/2020 9:47:38 PM:

```
[Pipeline] // stage
[Pipeline] stage
[Pipeline] { (Build and Push Docker Image to Registry)
[Pipeline] container
[Pipeline] {
[Pipeline] sh
+ pwd
+ /kaniko/executor --context '/home/jenkins/agent/workspace/Curriculum Vitae' --destination prengineer/cvjp:latest
[36mINFO[0m[0000] Retrieving image manifest alpine:latest      
[36mINFO[0m[0001] Retrieving image alpine:latest from registry index.docker.io 
[36mINFO[0m[0001] Built cross stage deps: map[]                
[36mINFO[0m[0001] Retrieving image manifest alpine:latest      
[36mINFO[0m[0001] Returning cached image manifest              
[36mINFO[0m[0001] Executing 0 build triggers                   
[36mINFO[0m[0001] Building stage 'alpine:latest' [idx: '0', base-idx: '-1'] 
[36mINFO[0m[0001] Unpacking rootfs as cmd RUN apk --no-cache --update     add apache2     apache2-ssl     && mkdir /app requires it. 
[36mINFO[0m[0001] LABEL maintainer="Jorge Pabón <pianistapr@hotmail.com>" description="Alpine + Apache2 with my CV in a JavaScript web app." 
[36mINFO[0m[0001] Applying label maintainer=Jorge Pabón <pianistapr@hotmail.com> 
[36mINFO[0m[0001] Applying label description=Alpine + Apache2 with my CV in a JavaScript web app. 
[36mINFO[0m[0001] RUN apk --no-cache --update     add apache2     apache2-ssl     && mkdir /app 
[36mINFO[0m[0001] Initializing snapshotter ...                 
[36mINFO[0m[0001] Taking snapshot of full filesystem...        
[36mINFO[0m[0004] Cmd: /bin/sh                                 
[36mINFO[0m[0004] Args: [-c apk --no-cache --update     add apache2     apache2-ssl     && mkdir /app] 
[36mINFO[0m[0004] Running: [/bin/sh -c apk --no-cache --update     add apache2     apache2-ssl     && mkdir /app] 
fetch https://dl-cdn.alpinelinux.org/alpine/v3.18/main/aarch64/APKINDEX.tar.gz
fetch https://dl-cdn.alpinelinux.org/alpine/v3.18/community/aarch64/APKINDEX.tar.gz
(1/8) Installing libuuid (2.38.1-r8)
(2/8) Installing apr (1.7.4-r0)
(3/8) Installing libexpat (2.5.0-r1)
(4/8) Installing apr-util (1.6.3-r1)
(5/8) Installing pcre2 (10.42-r1)
(6/8) Installing apache2 (2.4.57-r3)
Executing apache2-2.4.57-r3.pre-install
(7/8) Installing openssl (3.1.2-r0)
(8/8) Installing apache2-ssl (2.4.57-r3)
Executing apache2-ssl-2.4.57-r3.post-install
Executing busybox-1.36.1-r2.trigger
OK: 17 MiB in 23 packages
[36mINFO[0m[0006] Taking snapshot of full filesystem...        
[36mINFO[0m[0008] COPY ./App /app                              
[36mINFO[0m[0009] Taking snapshot of files...                  
[36mINFO[0m[0009] EXPOSE 80                                    
[36mINFO[0m[0009] Cmd: EXPOSE                                  
[36mINFO[0m[0009] Adding exposed port: 80/tcp                  
[36mINFO[0m[0009] Using files from context: [/home/jenkins/agent/workspace/Curriculum Vitae/entrypoint.sh] 
[36mINFO[0m[0009] ADD entrypoint.sh /                          
[36mINFO[0m[0009] Taking snapshot of files...                  
[36mINFO[0m[0009] RUN ["chmod", "+x", "/entrypoint.sh"]        
[36mINFO[0m[0009] Cmd: /bin/chmod                              
[36mINFO[0m[0009] Args: [+x /entrypoint.sh]                    
[36mINFO[0m[0009] Running: [/bin/chmod +x /entrypoint.sh]      
[36mINFO[0m[0009] Taking snapshot of full filesystem...        
[36mINFO[0m[0012] ENTRYPOINT ["/entrypoint.sh"]                
[36mINFO[0m[0012] Pushing image to prengineer/cvjp:latest      
[36mINFO[0m[0014] Pushed index.docker.io/prengineer/cvjp@sha256:b7775ec1e85a21c8284f0d59e66fa8f8ea37ca12bcb89ed4f19e644fabbbf4b2
```

**Comment by PREngineer** [comment]
Hi Aaron, I think I have found the issue.

Turns out that in my cluster, all of my Pis were running aarch64/arm64/armv8 **except** the master node.  That one was running armv7l.  I just realized the stupid mistake.  My apologies for wasting your time.  I just have to rebuild my cluster, making sure all of the nodes are arm64.

It makes sense if I think that the cluster tried to spin up the pod in the master node and it couldn't find the proper architecture for it.  I assume there is no armv7l version of the image.
